llama_stack/distribution/server/server.py

```diff
@@ -484,5 +485,47 @@ def extract_path_params(route: str) -> List[str]:
     return params
 
 
+def is_flag_set(flag: str, args: List[str], default_args: argparse.Namespace) -> bool:
+    """
+    Checks if a specific flag is set in the provided arguments.
+
+    Args:
+        flag (str): The name of the flag to check.
+        args (List[str]): The list of arguments to check against.
+        default_args (argparse.Namespace): The default arguments to compare with.
+
+    Returns:
+        bool: True if the flag is set in the provided arguments, False otherwise.
+    """
+    return vars(default_args)[flag] != vars(args)[flag]
+
+
+def return_flag_value(
+    flag: str,
+    args: argparse.Namespace,
+    default_args: argparse.Namespace,
+    config_file_value: Optional[str],
+    env_var: Optional[str] = None,
+) -> str:
+    """
+    Determine the final value for a flag based on priority:
+    1. Environment Variable (ENV_VAR) → Highest priority
+    2. Command-line Flag (--option)   → Medium priority
+    3. Config File Option             → Lowest priority
+    4. Default Argument               → Fallback when nothing else is set
+    """
+
+    # Get values from different sources
+    env_value = os.environ.get(env_var) if env_var else None
+    cli_value = vars(args).get(flag)
+    default_value = vars(default_args).get(flag)
+
+    # Check if the CLI flag was explicitly set
+    user_set_flag = is_flag_set(flag, args, default_args)
+
+    # Apply priority order: ENV > CLI > Config > Default
+    return env_value or (cli_value if user_set_flag else config_file_value or default_value)
```

Comment:
Since you know the precedence order in advance, I think you should return each value in order once you know it's set.

```
env_value = ...
if env_value:
   return env_value

if user_set_flag: ...

if config_file_value: ...

return default_value
```

(Sadly Python is not lazy...)

Reply:
It's more readable too! Thanks!